src/briefcase/platforms/macOS/__init__.py

```diff
@@ -765,93 +863,234 @@ def package_app(
             if notarize_app is None:
                 notarize_app = True
 
-            self.logger.info(f"Signing app with identity {identity_name}...")
-
-            if notarize_app:
-                team_id = self.team_id_from_identity(identity_name)
+            self.logger.info(f"Signing app with identity {identity.name}...")
 
         self.sign_app(app=app, identity=identity)
 
+        if app.packaging_format == "zip":
+            self.package_zip(
+                app,
+                notarize_app=notarize_app,
+                identity=identity,
+            )
+
+        elif app.packaging_format == "pkg":
+            self.package_pkg(
+                app,
+                notarize_app=notarize_app,
+                identity=identity,
+            )
+
+        else:  # Default packaging format is DMG
+            self.package_dmg(
+                app,
+                notarize_app=notarize_app,
+                identity=identity,
+            )
+
+    def package_zip(
+        self,
+        app: AppConfig,
+        notarize_app: bool,
+        identity: SigningIdentity,
+    ):
+        """Package an .app bundle in a zip file."""
         dist_path: Path = self.distribution_path(app)
 
-        if app.packaging_format == "app":
-            if notarize_app:
-                self.logger.info(
-                    f"Notarizing app using team ID {team_id}...",
-                    prefix=app.app_name,
+        if notarize_app:
+            self.logger.info(
+                f"Notarizing app using team ID {identity.team_id}...",
+                prefix=app.app_name,
+            )
+            self.notarize(self.binary_path(app), identity=identity)
+
+        with self.input.wait_bar(f"Archiving {dist_path.name}..."):
+            self.tools.shutil.make_archive(
+                dist_path.with_suffix(""),
+                format="zip",
+                root_dir=self.binary_path(app).parent,
+                base_dir=self.binary_path(app).name,
+            )
+
+    def package_pkg(
+        self,
+        app: AppConfig,
+        notarize_app: bool,
```

Comment:
True - that's a leftover of a feature that I started looking at, then backed out of. 

It is *possible* to sign and notarise installers; however, the signing process requires a separate signing identity to the app distribution signing identity. 

This is a solvable problem - one that we should open as a feature ticket as soon as this PR lands - but it's a little fiddly, and AFAICT, it's not needed to use the installer in practice

My inclination is to leave the argument in, even though it isn't used, because it *should* be used; but I won't argue too hard if you think it should be removed.

Reply:
OK, let's keep it but leave a comment in the code explaining the situation.

To test this, we agreed that you'll sign a pkg and send it to me to make sure I can install it.

Reply:
I had the sudden realization that I have a son with an M1 laptop (🤦), so I’ve just tested the installer on his machine. It works, but it triggers the “unknown developer” handling. It’s installable, but you’ve got to do the “explicit allow” dance through the Settings panel. So, it would appear that notarisation is a necessary step. Any preferences on rolling that change into this PR, vs a follow up?

Reply:
I've finished testing on all three platforms and found no other problems, so let's start merging now and do it as a follow-up.

Reply:
I've approved the Xcode and VisualStudio PRs.

Reply:
PRs for the app templates have been updated with new stubs; they're now ready for review.

Reply:
> _So, it would appear that notarisation is a necessary step._

FWIW technically i don't think this is called "notarization"... that's a whole _other_ process with a whole _other_ developer certificate that you do before submitting to the app store. to get the installer to work without "unknown developer" issues (or at least none beyond right click -> install anyway) you can just run `pkgbuild` with the  `--sign "Developer ID Installer: Nasir Jones (XYZXYZXYZ)"`.

my understanding - which is not 100% but i did manage to get a briefcase app into an installer successfully - is that there may be certificate types that can be used for both packaging and notarizing but there are definitely certificate types that will only do one or the other.

<img width="812" alt="Screenshot 2024-06-04 at 1 08 27 AM" src="https://github.com/beeware/briefcase/assets/111080266/3f31bc27-5d31-47a0-b534-ed5f137e8600">

Reply:
Ok - technically *signing* is a required step; Notarization is an extra step that you might as well do once you're signing (especially given that we've got notarization ready to go). 

However, the key detail is that you can't use the "Developer ID Application" certificate to sign (and thus notarize) the `pkg` archive, so additional certificate/identity handling is required.

Reply:
[this issue](https://github.com/electron/notarize/issues/60) has some more of the details on the difference between signing and notarizing

Reply:
> _the key detail is that you can't use the "Developer ID Application" certificate to sign_

correct

> _Notarization is an extra step that you might as well do once you're signing_

it could very well be but tbh i'm not sure... as far as i can tell (and i could be wrong) to get an app notarized requires sending it to apple and getting something back from them. i don't know if there are any kind of limits on how many packages an app developer can sign but i'd be a bit wary of just automatically submitting to apple for notarization every time the package is built (at least not without fully understanding the apple notarization process).

Reply:
Yes, it involves sending code to Apple and getting a response back. Briefcase already manages this process.

And yes, there is a rate limit. You don't want to put notarisation in a CI process, but there's no issue with having lots of apps, as long as you don't try to notarise them all on the same day.